eda/eda_script.py

```diff
@@ -0,0 +1,111 @@
+# author: Maeve Shi
+#date: 2021-11-24
+
+"""
+This script performs exploratory data visualization on a given dataset
+
+Usage: eda/eda_script.py --data_path=<data_path> --outputfile=<outputfile>
+Options:
+--data_path=<data_path>           Specify the path of the input data.
+--outputfile=<outputfile>         Specify the place to save the images.
+"""
+
+# import relevant modules
+from datetime import date
+from altair.vegalite.v4.schema.core import LinearGradient
+from docopt import docopt
+import altair as alt
+
+import numpy as np
+import pandas as pd
+from altair_saver import save
+import os
+
+alt.data_transformers.enable('data_server')
+alt.renderers.enable('mimetype')
+
+opt = docopt(__doc__)
+
+#extract the file name
+
+name = os.path.splitext(os.path.split(opt["--data_path"])[1])[0]
+name_hist = '/'+ name + '_histogram.png'
+name_line = '/'+ name + '_line.png'
+
+
+def main(data_path, outputfile):
+    """
+    calls visualization function to create charts and bars from the dataset
+    and save them as png files at a specified location.
+    -----
+    data_path: str
+        path of the input dataset
+    outputfile: str
+        path to save the output images
+    Returns
+    -----
+    None
+    """
+    # read the dataset
+    data = pd.read_csv(data_path)
+    #creat path of the output img if the path doesn't exist
+    if not os.path.exists(os.path.dirname(outputfile)):
+        os.makedirs(os.path.dirname(outputfile))
+    #get the distribution of the target feature
+    histograms(data, outputfile)
+    #get the time series plot of the target feature
+    time_line(data, outputfile)
+
+
+def histograms(data, outputfile):
+    """
+    create histograms of given features to obtain the distribution 
+    to see the difference before and after the pandamic,
+    then save the figure as png file to the outputfile location.
+    -----
+    data: pandas dataframe
+    outputfile: str
+      path to save the output images.
+    Returns:
+    -----
+    None
+    """
+    #create histogram of given column/feature
+
+    hist = (
+        alt.Chart(data).mark_bar(opacity=0.8).encode(
+            x=alt.X('substance_use_total', bin=alt.Bin(maxbins=30)),
+            y='count()')
+    )
+
+    #save the file
+    save(hist, outputfile + name_hist)
+
+def time_line(data, outputfile):
```

Comment:
v minor point - you can probably call this function 'create_timeseries'

Reply:
yeah nice catch! Will delete the file and modify it